### DISCUSSION TITLE
Trying to run async code doesn't work

### DISCUSSION THREAD

**Discussion by alobaili** [SwiftFiddle/swiftfiddle-web#101] [Q&A] [answered]
I'm trying to start execution on an async context to support running async code like so:

```swift
@main
struct Main {
  static func main() async {
    await doSomething()
  }

  static func doSomething() async {}
}
```

but I get the following error on `@main`:
```
Swift version 5.5-dev (LLVM 75998a7b14501e1, Swift 38a8b008bc66b11)
Target: x86_64-unknown-linux-gnu
/[REDACTED]/main.swift:5:1: error: 'main' attribute cannot be used in a module that contains top-level code
@main
^
/[REDACTED]/main.swift:1:1: note: top-level code defined in this source file
import Glibc
^
*** No output. ***
```

I'm guessing there's some kind of import on `main.swift` that isn't visible on the editor that I can't remove to support this.

Is there a way around this to make experimenting with async code easy?

Thanks!

**Comment by kishikawakatsumi** [comment, marked as answer]
You're right. Swiftfiddle automatically add the code equivalent to @main in favor of simplicity.
Also Async/await is still actively changed and Linux support is behind that of macOS. so the following is a bit old, but it works.

```swift
import Foundation

struct Main {
  static func main() async {
    await doSomething()
  }

  static func doSomething() async {
    print("doSomething")
  }
}


Task.detached {
  await Main.main()
  exit(EXIT_SUCCESS)
}
RunLoop.main.run()
```

https://swiftfiddle.com/4e3ig5vph5d2vnl2biuhoj33ti

There are two important points.
The Swiftfiddle code is executed at the top level, so executing async function, you need to enclose it in `Task.detached()`.

The execution of the async method will return soon, so use RunLoop to wait for avoiding the whole execution finishes before await function finishes.
When the async method is finished, terminate the program to stop the wait.


The following is a sample of Async/Await that I wrote. Please use this as a reference.
https://async-await-in-swift.netlify.app/

**Reply by alobaili** [reply]
Thank you for your help!
